### PR TITLE
stacks no longer give you a pop up window when you alt click on one on the floor

### PR DESCRIPTION
**Pull request by Tlaltecuhtli**
:cl:
tweak: stacks no longer give you a pop up window when you alt click on one on the floor
/:cl:

its rather obnoxious when you try to grab something from a tile (example after an explosion) which has some giant sprites of rod/tiles/metal and you want toalt click to see the tile content but instead you have to pixel snipe else you get a pop up window

99% of the time you want to grab part of a stack you do it in hand or in the bag anyway which have no effects from this pr

### CODE REVIEW

**Comment by nemvar** [comment]
Also fixes being able to teleport stacks with xray.

File: code/game/objects/items/stacks/stack.dm [hunk 1]
```diff
@@ -355,6 +355,9 @@
 		. = ..()
 
 /obj/item/stack/AltClick(mob/living/user)
+	. = ..()
+	if(isturf(src.loc))
```
Comment (nemvar) [inline code/game/objects/items/stacks/stack.dm]: if(isturf(loc))

**Comment by ExcessiveUseOfCobblestone** [comment]
do you think it should output a message saying you need to be holding it to split it (in case someone is alt clicking it intentionally)?

**Comment by Tlaltecuhtli** [comment]
they will probably notice after 1-2 clicks that it doesnt give you a pop up, more chat bloat is bad

**Comment by nemvar** [comment]
You should probably update this tip.
```
+= "<span class='notice'>Alt-click to take a custom amount.</span>"

**Comment by optimumtact** [comment]
can you put a comment indicating why it returns early on turf locs? thanks

File: code/game/objects/items/stacks/stack.dm [hunk 2]
```diff
@@ -78,20 +78,20 @@
 	..()
```
Comment (SpaceManiac) [inline code/game/objects/items/stacks/stack.dm]: Needs `. = ..()` here and also what oranges said

**Review by SpaceManiac — CHANGES_REQUESTED** [review]

**Comment by terranaut1** [comment]
I fucking do this all the time to grab shit from the floor fuck you

**Pull request merged**